accounting/acct.proto

```diff
@@ -0,0 +1,156 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// A protocol buffer definition used for logging events reported to a
+// gNMI server. This file assumes the following nomenclature:
+//
+//   gNMI Accounting Server -> System
+//   gNxI accounting Server <-- messages -- System
+//
+// System is a router, switch, PC, ... which accepts a connection from a
+// gNMI accounting server to collect and record accounting Records.
+//
+// A gNMI accounting Server is a server which receives records from systems and
+// logs them in the format specified using messages defined in this file.
+//
+// The current version covers event recording of shell/vendor-CLI and gRPC
+// based connections.
+
+syntax = "proto3";
+
+package gnsi.accounting;
+
+option go_package = "github.com/openconfig/gnsi/accounting";
+
+// Session identifier that identifies the context in which service
+// was executed.
+message SessionInfo {
+  // address of the system sending the event, in most cases this address
+  // is the Loopback or Loopback equivalent on system which is sending
+  // the messages to the collector.
+  string system_address = 1;
+
+  // rest of the information for 5 tuple of the connection
+  uint32 layer4_proto = 2;
+  uint32 local_port = 3;
+  string remote_address = 4;
+  uint32 remote_port = 5;
+
+  // For multiplexing multiple channels over single 5 tuple
+  // - multiple CLI connections over single SSH
+  string channel_id = 6;
+}
+
+// Details of authentication - for gRPC or shell/vendor-CLI.
+message AuthenDetail {
+  // Identity string is used to identify the user that executed the
+  // command. For instance, it could be spiffe-id in the case of gRPC or
+  // unix-style user-name in the case of shell/vendor-CLI.
+  string identity = 1;
+
+  // Privilege level configured on the system.
+  uint32 privilege_level = 2;
+
+  // authentication status
+  enum AuthenStatus {
+    AUTHEN_STATUS_UNSPECIFIED = 0;
+    AUTHEN_STATUS_PERMIT = 1;
+    AUTHEN_STATUS_DENY = 2;
+  }
+  AuthenStatus status = 3;
+
+  // In case of STATUS_DENY, cause for the deny
+  string deny_cause = 4;
+}
+
+// Command details for shell/vendor-CLI
+message CommandService {
+  enum CmdServiceType {
+    CMD_SERVICE_TYPE_UNSPECIFIED = 0;
+    CMD_SERVICE_TYPE_SHELL = 1;
+    CMD_SERVICE_TYPE_CLI = 2;
+  }
+  CmdServiceType service_type = 1;
+
+  // The command that was executed.
+  string cmd = 2;
+
+  // Arguments to a command above.
+  repeated string cmd_args = 3;
+}
+
+// Command details for openconfig gNxI commands.
+message GrpcService {
```

Comment:
(nit, not sure, probably pedantic) Should this be `GRPCService` vs. `GrpcService`?

Reply:
I went the 'proto does camelcase' route. I really don't know though :)